src/components/SaveEventButton.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React from "react";
+import { Animated, Easing, StyleSheet } from "react-native";
+import LottieView from "lottie-react-native";
+import Touchable from "./Touchable";
+import heartAnimationLight from "../../assets/animations/save-event-light.json";
+import heartAnimationDark from "../../assets/animations/save-event-dark.json";
+
+type Props = {
+  active: boolean,
+  onDark: boolean,
+  onPress: boolean => void
+};
+
+type State = {
+  progress: Object
+};
+
+export default class SaveEventButton extends React.Component<Props, State> {
+  static defaultProps = {
+    active: false,
+    onDark: false
+  };
+
+  state = {
+    progress: new Animated.Value(0)
+  };
+
+  componentDidMount() {
+    if (this.props.active) {
+      this.state.progress = new Animated.Value(1);
```

Comment:
I'm confused. Should we be using `setState({ progress: new Animated.Value(1) });` here?

Reply:
We warn against that. This was a bit of a hack, I have now updated to use `getDerivedStateFromProps`.